tests/app/celery/test_process_ses_receipts_tasks.py

```diff
@@ -282,47 +291,94 @@ def test_ses_callback_should_update_multiple_notification_status_sent(
     assert send_mock.called
 
 
-def test_ses_callback_should_set_status_to_permanent_failure(client, mocker, sample_template, sample_notification):
+@pytest.mark.parametrize('status', [NOTIFICATION_DELIVERED, NOTIFICATION_SENDING, NOTIFICATION_SENT])
+def test_ses_callback_should_set_status_to_permanent_failure(
+    client,
+    mocker,
+    sample_template,
+    sample_notification,
+    sample_service_callback,
+    status,
+):
     send_mock = mocker.patch('app.celery.service_callback_tasks.send_delivery_status_to_service.apply_async')
 
     template = sample_template(template_type=EMAIL_TYPE)
     ref = str(uuid4())
     notification_id = sample_notification(
         template=template,
-        status='sending',
+        status=status,
         reference=ref,
     ).id
-    create_service_callback_api(service=template.service, url='https://original_url.com')
+    sample_service_callback(service=template.service, url='https://original_url.com')
 
-    assert get_notification_by_id(notification_id).status == 'sending'
-    assert process_ses_receipts_tasks.process_ses_results(ses_hard_bounce_callback(reference=ref))
-    assert get_notification_by_id(notification_id).status == 'permanent-failure'
-    assert get_notification_by_id(notification_id).status_reason == 'Failed to deliver email due to hard bounce'
+    assert process_ses_receipts_tasks.process_ses_results(ses_hard_bounce_callback(reference=ref)) is None
+    db_notification = get_notification_by_id(notification_id)
+    assert db_notification.status == NOTIFICATION_PERMANENT_FAILURE
+    assert db_notification.status_reason == 'Failed to deliver email due to hard bounce'
     assert send_mock.called
 
 
+@pytest.mark.parametrize('bounce_status', [NOTIFICATION_TEMPORARY_FAILURE, NOTIFICATION_PERMANENT_FAILURE])
+def test_ses_does_not_update_if_already_bounced(
+    client,
+    mocker,
+    sample_notification,
+    sample_template,
+    bounce_status,
+):
+    send_mock = mocker.patch('app.celery.service_callback_tasks.send_delivery_status_to_service.apply_async')
+
+    template = sample_template(template_type=EMAIL_TYPE)
+    ref = str(uuid4())
+    if bounce_status == NOTIFICATION_PERMANENT_FAILURE:
+        status_reason = 'Failed to deliver email due to hard bounce'
+    elif bounce_status == NOTIFICATION_TEMPORARY_FAILURE:
+        status_reason = 'Temporarily failed to deliver email due to hard bounce'
+    else:
+        raise NotImplementedError
+
+    notification_id = sample_notification(
+        template=template,
+        status=bounce_status,
+        status_reason=status_reason,
+        reference=ref,
+    ).id
+
+    assert process_ses_receipts_tasks.process_ses_results(ses_notification_callback(reference=ref)) is None
+    db_notification = get_notification_by_id(notification_id)
+    assert db_notification.status == bounce_status
+    assert db_notification.status_reason == status_reason
+    assert not send_mock.called
```

Comment:
https://docs.python.org/3.10/library/unittest.mock.html#unittest.mock.Mock.assert_not_called

```suggestion
    send_mock.assert_not_called()
```

Reply:
That is a preference and I see no reason to change this. The docs you linked have both. My preference is towards using assert for the visual queues.